File: fegen-example-gradle/web-api/src/controller/CustomEndpointControllerClient.ts
import {Contact, CreateRequest} from "../Entities";
import {RequestAdapter, stringHelper, isEndpointCallAllowed} from "@materna-se/fegen-runtime";

export class CustomEndpointControllerClient {

    private readonly requestAdapter: RequestAdapter;
    
    constructor(requestAdapter: RequestAdapter) {
        this.requestAdapter = requestAdapter;
    }
    
    public async createContact(body: CreateRequest): Promise<Contact>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/contacts/create`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as Contact;
        return responseObj;
        
        
    }
    
    public isCreateContactAllowed(): Promise<boolean> {
        const url = `/api/custom/contacts/create`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async createOrUpdateContact(userName: string, firstName: string, lastName: string, number: string, street: string, zip: string, city: string, country: string): Promise<Contact>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/contacts/createOrUpdate`;
    
        const params = {userName, firstName, lastName, number, street, zip, city, country};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as Contact;
        return responseObj;
        
        
    }
    
    public isCreateOrUpdateContactAllowed(): Promise<boolean> {
        const url = `/api/custom/contacts/createOrUpdate`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
}

File: fegen-example-gradle/kotlin-api/src/main/kotlin/de/materna/fegen/example/gradle/kotlin/api/controller/CustomEndpointControllerClient.kt
package de.materna.fegen.example.gradle.kotlin.api.controller

import de.materna.fegen.example.gradle.kotlin.api.Contact
import de.materna.fegen.example.gradle.kotlin.api.ContactDto
import de.materna.fegen.example.gradle.kotlin.api.CreateRequest
import de.materna.fegen.runtime.RequestAdapter
import de.materna.fegen.runtime.appendParams
import de.materna.fegen.runtime.isEndpointCallAllowed
import kotlin.Boolean
import kotlin.String
import kotlin.Suppress

class CustomEndpointControllerClient(
    private val requestAdapter: RequestAdapter
) {
    @Suppress("UNUSED")
    suspend fun createContact(body: CreateRequest): Contact {
        val url = """/api/custom/contacts/create""".appendParams()
        return requestAdapter.doSingleRequest<Contact, ContactDto, CreateRequest>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isCreateContactAllowed(): Boolean {
        val url = """/api/custom/contacts/create"""
        return isEndpointCallAllowed(requestAdapter.request, "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun createOrUpdateContact(
        userName: String,
        firstName: String,
        lastName: String,
        number: String,
        street: String,
        zip: String,
        city: String,
        country: String
    ): Contact {
        val url = """/api/custom/contacts/createOrUpdate""".appendParams(
            "userName" to userName,
            "firstName" to firstName,
            "lastName" to lastName,
            "number" to number,
            "street" to street,
            "zip" to zip,
            "city" to city,
            "country" to country
        )
        return requestAdapter.doSingleRequest<Contact, ContactDto>(
            url = url,
            method = "POST",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isCreateOrUpdateContactAllowed(): Boolean {
        val url = """/api/custom/contacts/createOrUpdate"""
        return isEndpointCallAllowed(requestAdapter.request, "POST", url)
    }
}
